server/web_ui/shared/static/shared.js

```diff
@@ -1,6 +1,52 @@
 // This is easier to have in JS than in WASM
 export function modal_hide_by_id(m) {
-    var elem = document.getElementById(m);
-    var modal = bootstrap.Modal.getInstance(elem);
+    const elem = document.getElementById(m);
+    const modal = bootstrap.Modal.getInstance(elem);
     modal.hide();
-};
\ No newline at end of file
+}
+
+function changeTag(node, newTag) {
+    const newNode = document.createElement(newTag);
+
+    [...node.attributes].map(({ name, value }) => {
+        newNode.setAttribute(name, value);
+    });
+
+    while (node.firstChild) {
+        newNode.appendChild(node.firstChild);
+    }
+
+    node.parentNode.replaceChild(newNode, node);
+    return newNode;
+}
+
+export function init_graphviz(graph_src) {
+    if (typeof Viz !== 'undefined') {
+        start_graphviz(graph_src);
+    } else {
+        let meta = document.querySelector("meta[src='/pkg/external/viz.js']");
+        if (meta) {
+            let script = changeTag(meta, "script");
+            script.addEventListener('load', () => {
+                start_graphviz(graph_src);
+            });
+        } else {
+            console.error("viz.js not found");
+        }
+    }
+}
+
+function start_graphviz(graph_src) {
+    Viz.instance().then(function(viz) {
+        const graphContainer = document.getElementById("graph-container");
+        if (graphContainer)
+            graphContainer.replaceChildren(viz.renderSVGElement(graph_src))
+    });
+}
+
+export function open_blank(content) {
```

Comment:
Can you please add a docstring saying it'll open a blank window and fill it with `content` - just so it's clear in an IDE without having to read the code 😄 
```suggestion
/** Opens a popup window `target=_blank` filled with `content`
* @param string content
*/
export function open_blank(content) {
```

Reply:
This doesn't propagate through the `extern "C" {` bindings for me, maybe a rustrover issue ?
I can add this to the rust function as well if you want

Reply:
Oh, it won't propagate, but documenting what functions do is sensible everywhere 😄